udata/features/oembed/api.py

```diff
@@ -45,18 +45,18 @@ def get(self):
                     return api.abort(400, 'Unknown dataset ID.')
             elif (item_kind == 'territory' and
                     current_app.config.get('ACTIVATE_TERRITORIES')):
+
                 try:
                     country, level, code, kind = item_id.split(':')
                 except ValueError:
                     return api.abort(400, 'Invalid territory ID.')
-                try:
-                    geozone = GeoZone.objects.get(
-                        id=':'.join([country, level, code]))
-                except GeoZone.DoesNotExist:
+                geoid = ':'.join((country, level, code))
```

Comment:
Maybe we should have a counterpart to `parse_geoid`?

Reply:
There is already but it's not useable here because this is not using the same logic than anywhere else:
- `fr` prefix is hardcoded, `level` is not the same as in `udata.core.spatial`
- `code` is including the `@validity` part
This code section is already using the `GeoZone.objects.resolve()` helper one line below, but this part is not factorizable.
(BTW, I factorized a `udata.spatial.geoids` helper module)